package/MDAnalysis/analysis/hbonds/wbridge_analysis.py

```diff
@@ -33,1527 +33,17 @@
-        # plus order of water bridge times the length of OH bond plus hydrogne bond distance
-        self.selection_distance = (2 * 2 + self.order * (2 + self.distance))
-
-        self.box = self.u.dimensions if self.pbc else None
-        self._residue_dict = {}
-        self._build_residue_dict(self.selection1)
-        self._build_residue_dict(self.selection2)
-        self._build_residue_dict(self.water_selection)
-
-        self._update_selection()
-
-        self.timesteps = []
-        if len(self._s1) and len(self._s2):
-            self._update_water_selection()
-        else:
-            logger.info("WaterBridgeAnalysis: no atoms found in the selection.")
-
-        logger.info("WaterBridgeAnalysis: initial checks passed.")
-
-        logger.info("WaterBridgeAnalysis: starting")
-        logger.debug("WaterBridgeAnalysis: donors    %r", self.donors)
-        logger.debug("WaterBridgeAnalysis: acceptors %r", self.acceptors)
-        logger.debug("WaterBridgeAnalysis: water bridge %r", self.water_selection)
-
-        if self.debug:
-            logger.debug("Toggling debug to %r", self.debug)
-        else:
-            logger.debug("WaterBridgeAnalysis: For full step-by-step debugging output use debug=True")
-
-        logger.info("Starting analysis (frame index start=%d stop=%d, step=%d)",
-                    self.start, self.stop, self.step)
-
-    def _donor2acceptor(self, donors, h_donors, acceptor):
-        if len(donors) == 0 or len(acceptor) == 0:
-            return []
-        if self.distance_type != 'heavy':
-            donors_idx = list(h_donors.keys())
-        else:
-            donors_idx = list(donors.keys())
-        result = []
-        # Code modified from p-j-smith
-        pairs, distances = capped_distance(self.u.atoms[donors_idx].positions,
-                                           self.u.atoms[acceptor].positions,
-                                           max_cutoff=self.distance, box=self.box,
-                                           return_distances=True)
-        if self.distance_type != 'heavy':
-            tmp_donors = [h_donors[donors_idx[idx]] for idx in pairs[:,0]]
-            tmp_hydrogens = [donors_idx[idx] for idx in pairs[:,0]]
-            tmp_acceptors = [acceptor[idx] for idx in pairs[:,1]]
-        else:
-            tmp_donors = []
-            tmp_hydrogens = []
-            tmp_acceptors = []
-            for idx in range(len(pairs[:,0])):
-                for h in donors[donors_idx[pairs[idx,0]]]:
-                    tmp_donors.append(donors_idx[pairs[idx,0]])
-                    tmp_hydrogens.append(h)
-                    tmp_acceptors.append(acceptor[pairs[idx,1]])
-
-        angles = np.rad2deg(
-            calc_angles(
-                self.u.atoms[tmp_donors].positions,
-                self.u.atoms[tmp_hydrogens].positions,
-                self.u.atoms[tmp_acceptors].positions,
-                box=self.box
-            )
+warnings.warn(
```

Comment:
So in the past we've been a little bit more cautious with stubs, specifically setting simplewarning through something like:

```python
with warnings.catch_warnings():
    warnings.simplefilter('always', DeprecationWarning)
    warnings.warn(("This module has been moved to "
                   "MDAnalysis.analysis.hydrogenbonds.wbridge_analysis "
                   "It will be removed in release 3.0"),
                  category=DeprecationWarning)

from ..hydrogenbonds.wbridge_analysis import WaterBridgeAnalysis
```

Whilst I'm not sure the order of the import matters so much, the use of filtering is probably useful in cases where people tend to just blanket remove all warnings. Pinging @orbeckst here who probably has more context on why this was the decision taken for stubs circa ~ v0.21.

Reply:
Sorry for the late (and unhelpful) reply: I don't remember and the more cautious approach by @IAlibay looks good to me.

Removal would be 2.0 (together with old h-bonds).